src/components/gamecanvas/GameCanvas.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FC, useEffect, useCallback, useRef, useState, ReactElement } from 'react';
+import cn from 'classnames';
+
+import GameMain from '../../game/gamemain';
+
+interface IGameCanvas {
+  className?: string;
+}
+
+const GameCanvas: FC<IGameCanvas> = ({ className }): ReactElement => {
+  const canvasRef = useRef<HTMLCanvasElement | null>(null);
+  const [ collisions, getCollisions ] = useState(0);
+  const [ score, getScore ] = useState(0);
+  const [ isGameOver, getGameOverStatus ] = useState(false);
+  const [ isGamePaused, getGamePauseStatus ] = useState(false);
```

Comment:
❗️
Это сеттеры, поэтому называть их нужно с префиксом set)
Плюс хорошим тоном будет использовать полное название поле в сеттере. Ниже привёл пример
```suggestion
  const [ collisions, setCollisions ] = useState(0);
  const [ score, getScore ] = useState(0);
  const [ isGameOver, setIsGameOverStatus ] = useState(false);
  const [ isGamePaused, setIsGamePauseStatus ] = useState(false);
```